homeassistant/components/vicare/__init__.py

```diff
@@ -115,6 +116,31 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     return unload_ok
 
 
+async def async_migrate_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Migrate old entry."""
+    if entry.version == 1:
+        if entry.minor_version == 1:
+            _LOGGER.debug(
+                "Migrating from version %s.%s", entry.version, entry.minor_version
+            )
+            hass.config_entries.async_update_entry(
+                entry,
+                minor_version=2,
+                data={
+                    CONF_CLIENT_ID: entry.data.get(CONF_CLIENT_ID),
+                    CONF_USERNAME: entry.data.get(CONF_USERNAME),
+                    CONF_PASSWORD: entry.data.get(CONF_PASSWORD),
```

Comment:
These all look required in the schema
`CONF_CLIENT_ID: entry.data[CONF_CLIENT_ID]`
...

Reply:
```suggestion
                    CONF_CLIENT_ID: entry.data[CONF_CLIENT_ID],
                    CONF_USERNAME: entry.data[CONF_USERNAME],
                    CONF_PASSWORD: entry.data[CONF_PASSWORD],
```